kol/onshape/download.py

```diff
@@ -905,19 +903,22 @@ def get_urdf_part(
         mass = config.default_part_mass
         uses_default_mass = True
 
-    # Move the mesh origin and dynamics from the part frame to the parent
-    # joint frame (since URDF expects this by convention).
+    # We update the STL mesh so that its origin is at the joint frame.
     mesh_origin = urdf.Origin.zero_origin()
-    center_of_mass = part_dynamic.center_of_mass_in_frame(stl_origin_to_part_tf)
+    center_of_mass = part_dynamic.center_of_mass_in_frame(stl_origin_to_link_tf)
 
     if uses_default_mass:
-        inertia_transformed = np.zeros((3, 3))
+        inertia_transformed = np.eye(3)
     else:
         inertia = part_dynamic.inertia_matrix
-        inertia_transformed = transform_inertia_tensor(inertia, np.matrix(stl_origin_to_part_tf[:3, :3]))
+        inertia_transformed = transform_inertia_tensor(inertia, np.matrix(stl_origin_to_link_tf[:3, :3]))
 
-    principal_axes = part_dynamic.principal_axes_in_frame(stl_origin_to_part_tf)
-    principal_axes_rpy = R.from_matrix(principal_axes).as_euler("xyz", degrees=False)
+    # Since we are already transforming the inertia tensor so that the
+    # principal axes are aligned with the link frame, we can just use the
+    # identity matrix for the principal axes.
+    # principal_axes = part_dynamic.principal_axes_in_frame(stl_origin_to_link_tf)
```

Comment:
I understand that this is the main change along with keeping the original sign of com_to_part_tf? I still don't see the math and how come the previous setup was working for most of the cases though if it's such a radical change.